sdk/core/core/src/az_span.c

```diff
@@ -230,13 +230,14 @@ AZ_NODISCARD az_result az_span_append(az_span destination, az_span source, az_sp
   int32_t const dest_length = az_span_length(destination);
   int32_t const dest_capacity = az_span_capacity(destination);
   int32_t const src_length = az_span_length(source);
-  if ((dest_capacity - dest_length) < src_length)
+  int32_t const dest_length_after_appending = dest_length + src_length;
+  if (dest_length_after_appending > dest_capacity)
```

Comment:
The way it was written before was resilient to integer overflow (if `dest_length + src_length > INT32_MAX` it becomes negative). We should revert this.